packages/app/src/api/visualization.js

```diff
@@ -1,46 +1,36 @@
-import { getInstance } from 'd2'
 import { getFieldsStringByType } from '../modules/fields'
 
-export const apiFetchVisualization = (type, id) =>
-    getInstance().then(d2 =>
-        d2.models[type].get(id, {
-            fields: getFieldsStringByType(type),
-        })
-    )
+export const apiFetchVisualization = (dataEngine, type, id) => {
+    const visualizationQuery = {
+        visualization: {
+            resource: `${type}s`, // XXX SUPER UGLY HACK will be fixed when the vis api PR is merged
```

Comment:
Should this maybe be `TODO:` so we can find it later?

Reply:
yes, but it's actually already addressed here: https://github.com/dhis2/data-visualizer-app/pull/527